### PR TITLE
Postfix Evaluation in C

### PR DESCRIPTION
**Pull request by HariniJeyaraman**
Postfix evaluation in C

Fixes ##1552

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Implemented Postfix Evaluation in C
- 

#### Languages Used:
-  C
- 

#### Files Added:
- Postfix_Evaluation.c
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by SakshiShreya** [comment]
Please complete the PR template properly.

**Comment by HariniJeyaraman** [comment]
@chandrikadeb7 updated the file with sample input and output at the end

**Comment by jainaman224** [comment]
Try to have a better PR title.

File: Postfix _Evaluation.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      {
+         op1=pop(&s);
+         op2=pop(&s);
+         optr=s.A[i];
+         x=evaluate(op2,op1,optr);
+         push(&s,x);
+      }
+   }
+
+printf("The result is :  ");
```
Comment (jainaman224) [inline Postfix _Evaluation.c]: Indentation is incorrect.

File: Postfix _Evaluation.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    else
+    {
+      ps->Top=ps->Top+1;
+      ps->A[ps->Top]=x;
+    }
+}
+
+int evaluate(int op1,int op2,char c)
+{
+   if(c=='+')
```
Comment (jainaman224) [inline Postfix _Evaluation.c]: Spaces around operators like `==`, `<`, `=` etc. Would be great and improves readability.

File: Postfix _Evaluation.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
+//Postfix Evaluation
```
Comment (jainaman224) [inline Postfix _Evaluation.c]: Fix file name and add this file to a folder. Have a look at existing file and folder names.

**Comment by SakshiShreya** [comment]
@HariniJeyaraman are you working on this PR?

**Comment by HariniJeyaraman** [comment]
> @HariniJeyaraman are you working on this PR?

Yes, I am working on it . I'm updating and making the changes that you requested

**Comment by HariniJeyaraman** [comment]
@SakshiShreya  @chandrikadeb7 @jainaman224 Please review this file. I've made the necessary updates and changes.

**Comment by HariniJeyaraman** [comment]
@SakshiShreya @jainaman224 @chandrikadeb7 Please review the code again.

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+   return op1 - op2;
+   else if(c == '*')
+   return op1 * op2;
+   else if(c == '/')
+   return op1/op2;
+}
+
+void print(struct stack s)
+{
+  int i;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation.

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 5]
```diff
@@ -0,0 +1,85 @@
+//Postfix Evaluation
+#include<stdio.h>
+#include<string.h>
+#include<ctype.h>
+struct stack
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Leave a space before struct

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+{
+   char A[100];
+   int Top;
+};
+void push(struct stack *ps,char x)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Leave a space before it.

**Comment by HariniJeyaraman** [comment]
@SakshiShreya @jainaman224 Please review the code as soon as possible. I have made the necessary changes that you requested

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+{
+    if(ps->Top == 99)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add a space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    printf("\nStack is Full\n");
+    else
+    {
+      ps->Top = ps->Top + 1;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add a space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      ps->A[ps->Top] = x;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add a space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c (hunk 6 above)
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    }
+}
+
+int evaluate(int op1,int op2,char c)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+   return op1 * op2;
+   else if(c == '/')
+   return op1/op2;
+}
+
+void print(struct stack s)
+{
+  int i;
+  for(i = 0 ; i <= s.Top ; i++)
+  printf("%d\t",s.A[i]);
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+int pop(struct stack *ps)
+{
+   int x;
+   if(ps->Top == -1)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+   printf("Stack is empty!Element cant be deleted\n");
+   else
+   {
+     x = ps->A[ps->Top];
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+     ps->Top = ps->Top-1;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space around ->

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+     return x;
+   }
+}
+
+int main()
+{
+  int op1,op2,x,i;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+  char optr;
+  struct stack s;
+  s.Top = -1;
+  printf("Enter the postfix exp to be evaluated \n");
+  gets(s.A);
+  for(i = 0 ; i < strlen(s.A) ; i++)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Remove space before ;

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+  {
+      if(isdigit(s.A[i]))
+      push(&s,s.A[i]-'0');
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after , and around the operator

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+      else if(s.A[i] == '+' || s.A[i] == '-' || s.A[i] == '*' || s.A[i] == '/')
+      {
+         op1 = pop(&s);
+         op2 = pop(&s);
+         optr = s.A[i];
+         x = evaluate(op2,op1,optr);
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+         push(&s,x);
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space after ,

**Comment by HariniJeyaraman** [comment]
@somya-kapoor Made all the changes requested. Please review again

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+   int Top;
+};
+
+void push(struct stack *ps, char x)
+{
+   if(ps -> Top == 99)
+   printf("\nStack is Full\n");
+   else
+   {
+   ps -> Top = ps -> Top + 1;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+struct stack
+{
+   char A[100];
+   int Top;
+};
+
+void push(struct stack *ps, char x)
+{
+   if(ps -> Top == 99)
+   printf("\nStack is Full\n");
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+   else
+   {
+   ps -> Top = ps -> Top + 1;
+   ps -> A[ps -> Top] = x;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+   }
+}
+
+int evaluate(int op1, int op2, char c)
+{
+   if(c == '+')
+   return op1 + op2;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+   else if(c == '-')
+   return op1 - op2;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+   else if(c == '*')
+   return op1 * op2;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+   else if(c == '/')
+   return op1 / op2;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+void print(struct stack s)
+{
+   int i;
+   for(i = 0 ; i <= s.Top ; i++)
+   printf("%d\t", s.A[i]);
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 28]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+int pop(struct stack *ps)
+{
+   int x;
+   if(ps->Top == -1)
+   printf("Stack is empty!Element cant be deleted\n");
+   else
+   {
+     x = ps -> A[ps -> Top];
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation(4 space indentation)

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 29]
```diff
@@ -0,0 +1,87 @@
...
+     ps -> Top = ps -> Top - 1;
+     return x;
+   }
+}
+
+int main()
+{
+  int op1, op2, x, i;
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Please correct the indentation(4 space indentation)

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+   print(s);
+   printf("\n");   
+}
+
+/*Input - Output Sample
+Enter the postfix exp to be evaluated 
+2 3 1 * + 9 -
+The result is :  -4
+*/
+
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Remove the line

Comment (HariniJeyaraman) [reply Postfix_Evaluation/Postfix _Evaluation.c]: @somya-kapoor Removed the line

File: Postfix_Evaluation/Postfix _Evaluation.c (hunk 29 above)
Comment (HariniJeyaraman) [reply Postfix_Evaluation/Postfix _Evaluation.c]: @somya-kapoor Corrected the indentation to 4 spaces

File: Postfix_Evaluation/Postfix _Evaluation.c [hunk 31]
```diff
@@ -0,0 +1,86 @@
...
+{
+    int i;
+    for(i = 0 ; i <= s.Top ; i++)
+       printf("%d\t", s.A[i]);
+}
+
+int pop(struct stack *ps)
+{
+    int x;
+    if(ps->Top == -1)
```
Comment (somya-kapoor) [inline Postfix_Evaluation/Postfix _Evaluation.c]: Add space around ->

**Comment by somya-kapoor** [comment]
Good work. After making the changes, please squash the commits.

**Comment by HariniJeyaraman** [comment]
@somya-kapoor Made all changes. Should I make a new pull request after squashing the commits ?

**Comment by HariniJeyaraman** [comment]
@somya-kapoor Please review the code. Thanks

**Comment by somya-kapoor** [comment]
@HariniJeyaraman Please squash the commits. For help: http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

**Comment by HariniJeyaraman** [comment]
@somya-kapoor made a single commit for the file Postfix_Evaluation.

**Comment by HariniJeyaraman** [comment]
@somya-kapoor Please review again. Made into a single commit !

**Comment by HariniJeyaraman** [comment]
@jainaman224 Please review the code. Made all the changes requested by you as well as @somya-kapoor .

**Pull request merged**